File: 运营平台/app/routes/mysql.js
const mysql = require('mysql');
const connection = mysql.createConnection({
    host: 'localhost',
    user: 'root',
    password: '',
    database: 'mysql',
    port: 3306
});
function DataBase() {
    this.added = function (str, cb) {
        var sql = 'INSERT INTO test(渠道号) VALUES(?)';
        connection.query(sql, [str], function (err, result) {

            if (err) {
                cb(false);
                console.log(err);
                return;
            }
            else if (result) {
                cb(true);
                console.log("添加成功");
            }
            else {
                cb(false);
                console.log("添加失败");
            }

        })
    }
    this.init = function (tbname, cb) {
        var exsit="SELECT table_name FROM information_schema.TABLES WHERE table_name = " +  "'"+tbname+"'"+""
        var sql = 'select * from ' + tbname;
        //var creat="CREATE TABLE  " + tbname + " (渠道号 varchar(255),渠道名 varchar(255),素材 varchar(255),消耗 varchar(255),曝光 varchar(255),点击 varchar(255),点击率 varchar(255),新增（授权前） varchar(255),新增（授权后） varchar(255),授权转化 varchar(255),开局账号数 varchar(255),开局率 varchar(255),人均盘数 varchar(255),日期 varchar(255))"

        connection.query(exsit, function (err, result) {
            if (err) {
                
                console.log(err);
                return;
            }
            else if (result.length) {
                connection.query(sql, function (err, result) {
                    if (err) {
                        cb(false);
                        console.log(err);
                        return;
                    }
                    else if (result) {
                        cb(true,result)
                        console.log("init");
    
                    }
                    else {
                        cb(false);
                        console.log("initfalse");
                    }
    
                })
               
               
            }
            else{
                console.log("没有此表");
            }
        });
    };
    this.create=function(tbname){
        
    }
        this.find = function (pname, date, qudao, cb) {
            var length = qudao.length;
            var datalist = [];
            var sql;
            if (pname && date && !qudao) {
                sql = 'select * from ' + pname + ' where  日期=?';
                datalist = [date];
            }
            if (pname && date && qudao) {
                if (length == 1) {
                    sql = 'select * from ' + pname + ' where  日期=? and  渠道号=?';
                }
                else {
                    sql = 'select * from ' + pname + ' where   日期=? and  (渠道号=?';
                }

                datalist = [date, qudao[0]];
                for (var i = 1; i < length; i++) {
                    datalist.push(qudao[i]);
                    if (i == length - 1) {
                        sql += "or 渠道号=?)"
                        break;
                    }
                    sql += "or 渠道号=?";
                }

            }
            if (pname && !date && qudao) {
                if (length == 1) {
                    sql = 'select * from ' + pname + ' where   渠道号=?';
                }
                else {
                    sql = 'select * from ' + pname + ' where  (渠道号=?';
                }
                datalist = [qudao[0]];
                for (var i = 1; i < length; i++) {
                    datalist.push(qudao[i]);
                    if (i == length - 1) {
                        sql += "or 渠道号=?)"
                        break;
                    }
                    sql += "or 渠道号=?";
                }

            }
            if (pname && !date && !qudao) {
                sql = 'select * from ' + pname + '';
            }

            connection.query(sql, datalist, function (err, result) {
                if (err) {
                    cb(false);
                    console.log(err);
                    return;
                }
                else if (result) {
                    cb(true, result);

                }
                else {
                    cb(false);
                    console.log("错误");
                }

            })
        }

    connection.connect(function (err) {
        if (err) {
            console.log("数据库连接失败" + err);
        }
        else {
            console.log("数据库连接成功");
        }
    });
   this.create()
}

module.exports = new DataBase();

File: 运营平台/app/mysql.py
# coding=utf-8
import pymysql
import sys
reload(sys)
sys.setdefaultencoding('utf8')
conn = pymysql.connect(
    host="localhost", user="root", password="", database="mysql", port=3306, charset="utf8")
cursor = conn.cursor()


def addMPDB(consume,img,click,proname, probability, mp, date, channle,exposure):
    channlelist=channle.split("MP")
    if len(channlelist)>1:
        channle=channlelist[1]
    else:
        channle=channlelist[0]
    print channle
    table = "CREATE TABLE  " + proname + " (项目名 varchar(255),渠道号 varchar(255),渠道名 varchar(255),素材 varchar(255),消耗 varchar(255),曝光 varchar(255),点击 varchar(255),点击率 varchar(255),新增（授权前） varchar(255),新增（授权后） varchar(255),授权转化 varchar(255),开局账号数 varchar(255),开局率 varchar(255),人均盘数 varchar(255),日期 varchar(255))ENGINE=InnoDB DEFAULT CHARSET=utf8"
    sql = "INSERT INTO  "+proname+" (项目名,素材,日期,渠道号,渠道名,消耗,曝光,点击,点击率) VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s)" % (
      '"'+proname+'"','"'+img+'"', '"'+date+'"', '"'+channle+'"','"' + mp + '"', '"' + consume + '"', '"' + exposure + '"', '"' + click + '"', '"' + probability + '"')
    exsit = "SELECT table_name FROM information_schema.TABLES WHERE table_name =" +  "'"+proname+"'"+""
    try:
        if  cursor.execute(exsit)==1:
            cursor.execute(sql)
            conn.commit()
            
            print "success1"
        else :
            cursor.execute(table)
            cursor.execute(sql)
            conn.commit()
            print "success2"

    except Exception, e:
        print e

def updateMagicDB(proname,date,newsb,newzc,conversion,channel):
    # WHERE 日期=%s and 渠道号=%s
    sql=  "UPDATE " + proname + " SET 新增（授权前）=%s, 新增（授权后）=%s ,授权转化=%s WHERE 日期=%s and 渠道号=%s " %('"' + newsb + '"', '"' + newzc + '"', '"' + conversion + '"', '"' + date + '"','"' + channel + '"')
    try:
        cursor.execute(sql)
        conn.commit()
        print "update授权"
    except:
        conn.rollback()

def updateNewGamePlnum(proname,date,newgame,channel):
    # WHERE 日期=%s and 渠道号=%s
    print newgame,date
    sql=  "UPDATE " + proname + " SET 开局账号数=%s WHERE 日期=%s  and 渠道号=%s " %('"' + newgame + '"', '"' + date + '"', '"' + channel + '"')
    try:
        cursor.execute(sql)
        conn.commit()
        print "update新注册"
    except:
        conn.rollback()        
def updatePlayNum(proname,date,ps,channel):
    sql=  "UPDATE " + proname + " SET 人均盘数=%s WHERE 日期=%s and 渠道号=%s " %('"' + ps + '"', '"' + date + '"', '"' + channel + '"')        
    try:
        cursor.execute(sql)
        conn.commit()
        print "update盘数"
    except:
        conn.rollback()

def emptytable(tablename):
    sql = "delete from  "+tablename+""
    exsit = "SELECT table_name FROM information_schema.TABLES WHERE table_name =" +  "'"+tablename+"'"+""
    try:
        if cursor.execute(exsit)==1:
            cursor.execute(sql)
            conn.commit()
            print "dele success"
        else:
            return 
    except Exception, e:
        print e
def updateKaiJuLv(tablename):
    sql="SELECT 开局账号数,新增（授权后）,日期,渠道号  FROM "+tablename+" WHERE  新增（授权后） IS NOT NULL and 开局账号数 IS NOT NULL"
    
    try:
        cursor.execute(sql)
        results = cursor.fetchall()
        # print results
        if len(results)==0:
            return
        for item in results:
            
            if int(item[1])==0:
                rare="0%"
                updatesql="UPDATE " + tablename + " SET 开局率=%s WHERE 日期=%s and 渠道号=%s" %('"' + rare + '"','"' + item[2] + '"','"' + item[3] + '"')
                cursor.execute(updatesql)
                conn.commit()
                print "update 开局率"
            
            else:
                rare=str(int(round(float(item[0])/float(item[1]), 2)*100))+"%"
                updatesql="UPDATE " + tablename + " SET 开局率=%s WHERE 日期=%s and 渠道号=%s" %('"' + rare + '"','"' + item[2] + '"','"' + item[3] + '"')
                cursor.execute(updatesql)
                conn.commit()
                print "update 开局率"

        
    except Exception, e:
        print e

# addMPDB("a","aa","aaa","aaaaa", "干瞪眼", "--", "fff", "MP45114", "qqq")
# insertdict("aaaa","bbbbbbbb")
# addMagicDB("干瞪眼1","2019-02-20","10","10","100%","MP45098")
# updateNewGamePlnum("干瞪眼","2019-02-20","111","45116")
# updateKaiJuLv("干瞪眼")
